src/Schema/Column/BigIntColumnSchema.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Schema\Column;
+
+use Yiisoft\Db\Expression\ExpressionInterface;
+use Yiisoft\Db\Schema\SchemaInterface;
+
+use const PHP_INT_MAX;
+use const PHP_INT_MIN;
+
+class BigIntColumnSchema extends AbstractColumnSchema
+{
+    public function __construct(
+        string $type = SchemaInterface::TYPE_BIGINT,
+        string|null $phpType = SchemaInterface::PHP_TYPE_STRING,
+    ) {
+        parent::__construct($type, $phpType);
+    }
+
+    public function dbTypecast(mixed $value): int|string|ExpressionInterface|null
+    {
+        return match (gettype($value)) {
+            'string' => $value === '' ? null : (
+                $value <= PHP_INT_MAX && $value >= PHP_INT_MIN
+                    ? (int) $value
+                    : $value
+            ),
+            'NULL' => null,
+            'integer' => $value,
+            'boolean' => $value ? 1 : 0,
+            default => $value instanceof ExpressionInterface ? $value : (
+                ($val = (string) $value) <= PHP_INT_MAX && $val >= PHP_INT_MIN
+                    ? (int) $val
+                    : $val
+            ),
+        };
+    }
+
+    public function phpTypecast(mixed $value): string|null
```

Comment:
```suggestion
    public function phpTypecast(mixed $value): string|int|null
```

Can we make it?

Reply:
I thought about doing this before, but then I changed my mind.

For PHP there is no difference between `string` and `int` numbers, but supporting both types may be cause of cases when it is necessary to do more checks on whether the value is `string` or `int` or every time cast value to `string`.

The value from the database is assumed to be a large number and needs to be worked with as a large number, even if it is within the limits of the `int` type.